### PR TITLE
Refs [TASK] [36117] Lấy dữ liệu thông tin tướng

### PR DESCRIPTION
**Pull request by hhd0307**
## Related Tickets
- [#Task 36117: Lấy dữ liệu thông tin tướng](https://edu-redmine.sun-asterisk.vn/issues/36117)
## WHAT
- Get data detail hero
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/hero_01/data/model/HeroDetail.kt [hunk 1]
```diff
@@ -0,0 +1,77 @@
+package com.sun.hero_01.data.model
+
+data class HeroDetail(
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/hero_01/data/model/HeroDetail.kt:3]:
```suggestion
data class Hero(
```

File: app/src/main/java/com/sun/hero_01/data/source/remote/fetchdata/ParseJson.kt [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            getInt(HeroDetailEntry.STATS_HP),
+            getInt(HeroDetailEntry.STATS_MOVESPEED),
+            getInt(HeroDetailEntry.STATS_ARMOR),
+            getInt(HeroDetailEntry.STATS_ATTACK_RANGE),
+            getInt(HeroDetailEntry.STATS_HP_REGEN),
+            getInt(HeroDetailEntry.STATS_ATTACK_DAME),
+            getDouble(HeroDetailEntry.STATS_ATTACK_SPEED).toFloat(),
+            getDouble(HeroDetailEntry.STATS_SPEELBLOCK).toFloat()
+        )
+
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/hero_01/data/source/remote/fetchdata/ParseJson.kt]: remove blank

File: app/src/main/java/com/sun/hero_01/data/source/remote/fetchdata/ParseJson.kt [hunk 3]
```diff
@@ -0,0 +1,101 @@
+package com.sun.hero_01.data.source.remote.fetchdata
+
+import com.sun.hero_01.data.model.*
+import com.sun.hero_01.utils.KeyEntityType
+import org.json.JSONObject
+
+class ParseJson {
+
+    fun heroDetailParseJson(jsonObject: JSONObject) = jsonObject.run {
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/hero_01/data/source/remote/fetchdata/ParseJson.kt]: tương tự đặt lại tên các fun ở dưới e
```suggestion
    fun onParseJsonToHero(jsonObject: JSONObject) = jsonObject.run {
```

File: app/src/main/java/com/sun/hero_01/ui/detail/DetailFragment.kt [hunk 4]
```diff
@@ -1,4 +1,52 @@
...
+import com.sun.hero_01.base.BaseFragment
+import com.sun.hero_01.data.model.HeroDetail
+import com.sun.hero_01.data.source.HeroRepository
+import com.sun.hero_01.data.source.remote.HeroRemoteDataSource
+
+class DetailFragment : BaseFragment(), DetailContract.View {
+
+    override val layoutResourceId = R.layout.fragment_detail
+
+    private lateinit var idHero: String
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/hero_01/ui/detail/DetailFragment.kt]: khởi tạo dạng nullable

File: app/src/main/java/com/sun/hero_01/ui/detail/DetailFragment.kt [hunk 5]
```diff
@@ -1,4 +1,52 @@
...
+
+    companion object {
+        private const val ARGUMENT_HERO_ID = "ARGUMENT_HERO_ID"
+
+        fun newInstance(heroName: String): DetailFragment{
+            val args = Bundle()
+            val fragment = DetailFragment()
+            fragment.arguments = args
+            args.putString(ARGUMENT_HERO_ID, heroName)
+            return fragment
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/hero_01/ui/detail/DetailFragment.kt]: dùng scope function và thu gọn lại code

File: app/src/main/java/com/sun/hero_01/ui/detail/DetailPresenter.kt [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        repository?.getHeroDetails(id, object:
+            OnFetchDataJsonListener<HeroDetail> {
+            override fun onSuccess(data: HeroDetail) {
+                view?.loadHeroDetailOnSuccess(data)
+            }
+
+            override fun onError(exception: Exception?) {
+                view?.onError(exception)
+            }
+
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/hero_01/ui/detail/DetailPresenter.kt]: remove blank

File: app/src/main/java/com/sun/hero_01/data/model/HeroDetail.kt (hunk 1 above)
Comment (hhd0307) [reply app/src/main/java/com/sun/hero_01/data/model/HeroDetail.kt:3]: chỗ này e đặt HeroDetail để phân biệt với model Hero ở màn hình khác, Hero sẽ lấy ít thông tin hơn a.

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**